dfastmi/batch/AnalyserAndReporterWaqua.py

```diff
@@ -96,180 +86,282 @@ def analyse_and_report_waqua(
-    m = u0temp[:, 1].astype(int) - 1
-    n = u0temp[:, 2].astype(int) - 1
-    u0temp = u0temp[:, 0]
-    h0temp = DataTextFileOperations.read_waqua_xyz(files[1])
-    u1temp = DataTextFileOperations.read_waqua_xyz(files[2])
-
-    if reduced_output:
-        firstm = min(m)
-        firstn = min(n)
-    else:
-        firstm = 0
-        firstn = 0
-    szm = max(m) + 1 - firstm
-    szn = max(n) + 1 - firstn
-    szk = szm * szn
-    k = szn * m + n
-    u0 = numpy.zeros([szk])
-    h0 = numpy.zeros([szk])
-    u1 = numpy.zeros([szk])
-
-    u0[k] = u0temp
-    h0[k] = h0temp
-    u1[k] = u1temp
-
-    sz = [szm, szn]
-    u0 = u0.reshape(sz)
-    h0 = h0.reshape(sz)
-    u1 = u1.reshape(sz)
-
-    dzq = dfastmi.kernel.core.dzq_from_du_and_h(u0, h0, u1, ucrit)
-    if display:
-        ApplicationSettingsHelper.log_text("---")
-    return dzq, firstm, firstn
+        discriptions = ApplicationSettingsHelper.get_text("file_descriptions")
+        quantities = ["velocity-zeta.001", "waterdepth-zeta.001", "velocity-zeta.002"]
+        files = []
+        for i in range(3):
+            if self.display:
+                ApplicationSettingsHelper.log_text("input_xyz_name", dict={"name": discriptions[i]})
+            cifil = "xyz_" + quantities[i] + ".Q" + cblok + ".xyz"
+            if self.display:
+                print(cifil)
+            if not os.path.isfile(cifil):
+                ApplicationSettingsHelper.log_text("file_not_found", dict={"name": cifil})
+                ApplicationSettingsHelper.log_text("file_not_found", dict={"name": cifil}, file=self.report)
+                ApplicationSettingsHelper.log_text("end_program", file=self.report)
+                return False, files
+            else:
+                if self.display:
+                    ApplicationSettingsHelper.log_text("input_xyz_found", dict={"name": cifil})
+            files.extend([cifil])
+            if self.display:
+                ApplicationSettingsHelper.log_text("")
+                
+        return True, files
+    
+    def _read_files(self, stage : int, files : list) -> Tuple[numpy.ndarray, numpy.ndarray, numpy.ndarray]:
+        if self.display:
+            ApplicationSettingsHelper.log_text("input_xyz_read", dict={"stage": stage})
+            
+        u0temp = DataTextFileOperations.read_waqua_xyz(files[0], cols=(2, 3, 4))
+        h0temp = DataTextFileOperations.read_waqua_xyz(files[1])
+        u1temp = DataTextFileOperations.read_waqua_xyz(files[2])
+        
+        return u0temp, h0temp, u1temp
+
+    def _calculate_waqua_values(self, u0temp : numpy.ndarray, h0temp : numpy.ndarray, u1temp : numpy.ndarray) -> Tuple[int, int, numpy.ndarray]:
+        m = u0temp[:, 1].astype(int) - 1
+        n = u0temp[:, 2].astype(int) - 1
+        u0temp = u0temp[:, 0]
 
+        if self.reduced_output:
+            firstm = min(m)
+            firstn = min(n)
+        else:
+            firstm = 0
+            firstn = 0
```

Comment:
Should this not be in 2 seperate functions and passed as arguments to this method? Reduces the return values

```
def _calculate_waqua_values_min(self, u0temp : numpy.ndarray) -> int:
    m = u0temp[:, 1].astype(int) - 1
    if self.reduced_output:
        return min(m)
     else:
        return = 0

Reply:
done